### PR TITLE
Replace ResourceUtils

### PR DESCRIPTION
**Pull request by nickbp**
Hold on to your butts...

- Replaces `ResourceUtils`-based resource construction with a `ResourceBuilder`. Some code (particularly some evaluation stages) is still manually editing `Resource` objects but this cleans things up a ton regardless.
- Where applicable, some `ResourceUtils` functs were moved into `ResourceTestUtils` for use by tests, or into other `*Utils` classes.

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceCollectUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.mesos.Protos;
+import org.apache.mesos.Protos.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class encapsulates common methods for scanning collections of Resources.
+ */
+public class ResourceCollectUtils {
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceCollectUtils.java]: ResourceCollectionUtils is the best name for this class. Please?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java [hunk 2]
```diff
@@ -42,8 +44,9 @@
      */
     @Override
     public Collection<Resource> getPersistentVolumesToBeDestroyed(Offer offer) {
-        List<Resource> resources = offer.getResourcesList().stream().
-                filter(resource -> ResourceUtils.getPersistenceId(resource) != null).collect(Collectors.toList());
+        List<Resource> resources = offer.getResourcesList().stream()
+                .filter(resource -> resource.hasDisk() && resource.getDisk().hasPersistence())
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java]: nitpick: I'm a little worried about us now having this pattern of checking persistence getting copy pasted around.

Can it just become a public method of Resource?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+/**
+ * Constructs Mesos {@link Resource} protobufs.
+ */
+public class ResourceBuilder {
+
+    /** The name used for reserved network port resources. */
+    public static final String PORTS_RESOURCE_TYPE = "ports";
+    /** The name used for reserved storage/disk resources. */
+    public static final String DISK_RESOURCE_TYPE = "disk";
```
Comment (ArtRand) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java]: given that these types have associated types (SCALAR, RANGE, etc). Why not make these proper objects so we can validate input easier?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java (hunk 2 above)
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java]: Resource is a mesos protobuf. We should be able to resolve this once we've got our own classes for these types.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java (hunk 3 above)
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java]: I think that'll happen as we start moving away from passing around mesos protobufs internally

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java (hunk 2 above)
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java]: Ah, yeah.

Argh.

**Pull request merged**